### PR TITLE
feat: Added commonLabel to label resources applied via Configuration

### LINKED ISSUE #3555: commonLables in configuration.pkg for Composition Revisions Labels
### What problem are you facing?
we want to add commonLables in configuration.pkg, that can be used for composition revisions lables, in a automatic way
example:

```
---
apiVersion: pkg.crossplane.io/v1
kind: Configuration
metadata:
  name: test
spec:
  package: sop:v1.193.0
  commonLabels:
    version: v1.193.0
    channel: beta
```

when using ci and gitops tooling to package and deploy configuration packages it is not an option to update labels manually - nor is it feasible to use calculated hashes from compositionrevisions

### PR DESCRIPTION
**Pull request by akesser**
Signed-off-by: André Kesser <andre.kesser@dkb.de>

### Description of your changes

Added commonLabel property to Configuration.pkg to label applied resources. The idea is to be able to add labels that can later be used by compositionRevisionSelector -> matchLabels

Fixes #3555 

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

We tested the commonLabel attribute by applying a configuration with this attribute on a local kind cluster and observed the applied objects. We testet to create a new configuration with commonLabels, update a configuration without this attribute to add commonLabels and we tested to remove single and all commonLabels from a configuration.

Example: 
```yaml
kind: Configuration
metadata:
  name: test
spec:
  commonLabels:
    revision: v0.3.0
    test: "4"
  ignoreCrossplaneConstraints: true
  package: xpkg.upbound.io/crossplane-contrib/x-generation:v0.3.0
  packagePullPolicy: IfNotPresent
  revisionActivationPolicy: Automatic
  revisionHistoryLimit: 1
  skipDependencyResolution: true
```
With these compositions:
```yaml
apiVersion: apiextensions.crossplane.io/v1
kind: Composition
metadata:
  labels:
    example.cloud/provider: example
  name: compositeinstanceprofile.iam.aws.example.cloud
...
---
apiVersion: apiextensions.crossplane.io/v1
kind: Composition
metadata:
  labels:
    example.cloud/provider: example
  name: compositealias.kms.aws.example.cloud
...
```
leads to 
```yaml
apiVersion: apiextensions.crossplane.io/v1
kind: Composition
metadata:
  labels:
    example.cloud/provider: example
    revision: v0.3.0
    test: "4"
  name: compositeinstanceprofile.iam.aws.example.cloud
...
---
apiVersion: apiextensions.crossplane.io/v1
kind: Composition
metadata:
  labels:
    example.cloud/provider: example
    revision: v0.3.0
    test: "4"
  name: compositealias.kms.aws.example.cloud
...
```
etc.

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: internal/controller/pkg/revision/reconciler.go [hunk 1]
```diff
@@ -570,7 +572,24 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
...
+		}
+		labels := d.GetLabels()
+		if labels != nil {
+			for key, value := range commonLabels {
+				labels[key] = value
+			}
+		} else {
+			d.SetLabels(commonLabels)
+		}
+	}
```
Comment (hasheddan) [inline internal/controller/pkg/revision/reconciler.go:591]: Any reason not to do this as part of the establisher?

Comment (akesser) [reply internal/controller/pkg/revision/reconciler.go]: I moved the labeling not a method of the establisher

**Comment by ezgidemirel** [comment]
Hello @akesser , why do we need to add `commonLabels` field to CRDs other than configurations? My understanding was you want to propagate some labels to the resources created with a configuration. Am I missing something?

**Comment by akesser** [comment]
@ezgidemirel We need commonLabels on the resources to be able to use labelSelectors for CompositionRevisions. With labels compositionRevisionSelector.matchLabels can be used to select the compositions in a Claim so developer do not need to look up the calculated revision hash that needs to be used otherwise.

**Comment by haarchri** [comment]
@ezgidemirel  what you mean with with fields to CRDs other than configurations ? we tested configurations and added lables to crds - what "issue" you facing ?

**Comment by akesser** [comment]
Ah ok, we want commonLabel in ConfigurationSpec. ConfigurationSpec is the same as PackageSpec, from the Configuration/Package, a ProviderRevision object gets applied. The PackageRevision is then used in its own Reconciler to create the objects, therefore ProviderRevision needs the commonLabel attribute too. Providers share the same interface with Configuration called Package, ProviderRevisions share the same interface with ConfigurationRevison, called PackageRevison, therefore the labels appear there too.

**Comment by ezgidemirel** [comment]
Thanks @akesser for the explanation! The PR looks good to me. I think it's a good idea to have a common set of labels on resources that are coming from a configuration. It would definitely be useful with compositionSelectors as well.

**Review by ezgidemirel — APPROVED** [review]

**Comment by turkenh** [comment]
@akesser could you add some details under `How has this code been tested` section?

**Comment by akesser** [comment]
@turkenh I added details and Examples to the section

**Review by turkenh — APPROVED** [review]

**Pull request merged**